### PR TITLE
Fixed scroll to group item on Android + Windows

### LINKED ISSUE #8718: The ScrollTo method does not carry the element in a grouping
### Description

When a CollectionView is populated with grouped data, the ScrollTo method does not lead to the item indicated by the parameters.

When I invoke the method through:

```
     private void Button_Clicked(object sender, EventArgs e)
     {
          collectionView.ScrollTo(0, 4);
     }
```
hu3w1AZFwY

it is supposed to navigate to the first element of the fourth group, however, in the attached gif, you can see how nothing happens when you click on the button and the listing is on the first element, while when you scroll to the end of the listing, and press the button, it does not lead to the element it should be.

Attached is the project in Github: https://github.com/hprez21/CollectionViewScrollToIssue

### Steps to Reproduce

1.- Download the Github repo
2.- Run the application, and test the button with the text "Manual Scrolling".

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows, I was *not* able test on other platforms

### Affected platform versions

Android 10, Windows 10

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Foda**
### Description of Change

Fixed scrolling to a group via `CollectionView.ScrollTo(index, groupIndex)` not working on Windows or Android.

### Issues Fixed

Fixes #8718

### Demo

The demo below shows scrolling to the group letter "N"

https://github.com/dotnet/maui/assets/890772/af943949-e10d-416e-b928-d7114f1a48b7

https://github.com/dotnet/maui/assets/890772/51da9dac-f18a-4894-865a-30273433b0b4

### CODE REVIEW

File: src/Controls/tests/DeviceTests/Elements/CollectionView/CollectionViewTests.cs [hunk 1]
```diff
@@ -138,6 +139,91 @@ public async Task ItemsSourceDoesNotLeak()
...
+			};
+
+			await CreateHandlerAndAddToWindow<CollectionViewHandler>(collectionView, async handler =>
+			{
+				collectionView.ScrollTo(index: 4, groupIndex: 3); // Group 'c'
+
+				await Task.Delay(100);
+
+				// each group has 10 items, plus the 3 headers
+				Assert.True(lastVisibleItemIndex == 34);
```
Comment (jsuarezruiz) [inline src/Controls/tests/DeviceTests/Elements/CollectionView/CollectionViewTests.cs:207]: The test is failing on Android API_33:
image

**Review by PureWeen — APPROVED** [review]

**Pull request merged**